source/vof/handler.cc

```diff
@@ -0,0 +1,385 @@
+            {
+              has_vof_strategy = (plugin_names[name] == "vof interface") || has_vof_strategy;
+            }
+          AssertThrow(has_vof_strategy,
+                      ExcMessage("Volume of Fluid Interface Tracking requires that the 'vof interface' strategy be used for AMR"));
+        }
+        prm.leave_subsection();
+
+        AssertThrow(this->get_parameters().adaptive_refinement_interval <(1/this->get_parameters().CFL_number),
+                    ExcMessage("Volume of Fluid Interface Tracking requires that the AMR interval be less than 1/CFL_number"));
+      }
+
+    // Gather data
+
+    for (unsigned int f=0; f<n_vof_fields; ++f)
+      {
+        data.push_back(VoFField<dim>(this->introspection().variable("vof_"+vof_field_names[f]),
+                                     this->introspection().variable("vofN_"+vof_field_names[f]),
+                                     this->introspection().variable("vofLS_"+vof_field_names[f])));
+      }
+
+    // Do initial conditions setup
+    if (SimulatorAccess<dim> *sim_a = dynamic_cast<SimulatorAccess<dim>*>(vof_initial_conditions.get()))
+      sim_a->initialize_simulator (sim);
+    if (vof_initial_conditions.get())
+      {
+        vof_initial_conditions->parse_parameters (prm);
+        vof_initial_conditions->initialize ();
+      }
+
+  }
+
+  template <int dim>
+  unsigned int VoFHandler<dim>::get_n_fields() const
+  {
+    return n_vof_fields;
+  }
+
+  template <int dim>
+  const std::string VoFHandler<dim>::get_field_name(unsigned int field) const
+  {
+    Assert(field < n_vof_fields,
+           ExcMessage("Invalid field index"));
+    return vof_field_names[field];
+  }
+
+  template <int dim>
+  double VoFHandler<dim>::get_vof_epsilon() const
+  {
+    return vof_epsilon;
+  }
+
+  template <int dim>
+  const VoFField<dim> &VoFHandler<dim>::get_field(unsigned int field) const
+  {
+    Assert(field < n_vof_fields,
+           ExcMessage("Invalid field index"));
+    return data[field];
+  }
+
+  template <int dim>
+  unsigned int VoFHandler<dim>::get_vof_field(std::string composition_fieldname) const
+  {
+    if (vof_composition_map_index.count(composition_fieldname) ==0)
+      return n_vof_fields;
+    return vof_composition_map_index.at(composition_fieldname);
+  }
+
+  template <int dim>
+  void VoFHandler<dim>::do_vof_update ()
+  {
+    for (unsigned int f=0; f<n_vof_fields; ++f)
+      {
+        const unsigned int vof_block_idx = data[f].volume_fraction.block_index;
+        const unsigned int vofN_block_idx = data[f].reconstruction.block_index;
+
+        // Reset current base to values at beginning of timestep
+
+        // Due to dimensionally split formulation, use strang splitting
+        // TODO: Reformulate for unsplit (may require flux limiter)
+        bool update_from_old = true;
+        for (unsigned int dir = 0; dir < dim; ++dir)
+          {
+            // Update base to intermediate solution
+            if (!vof_dir_order_dsc)
```

Comment:
Please rename `vof_dir_order_dsc` to  `direction_order_descending`.

Reply:
Check whether this variable is stored during restart. Look in `particle_handler.h` at the function `serialize` for a template on how to do this.